sticker-utils/src/sent_proc.rs

```diff
@@ -21,20 +22,34 @@ impl<'a, W> SentProcessor<'a, W>
 where
     W: WriteSentence,
 {
-    pub fn new(pipeline: &'a Pipeline, writer: W, batch_size: usize, read_ahead: usize) -> Self {
+    pub fn new(
+        pipeline: &'a Pipeline,
+        writer: W,
+        batch_size: usize,
+        max_len: Option<usize>,
+        read_ahead: usize,
+    ) -> Self {
         assert!(batch_size > 0, "Batch size should at least be 1.");
         assert!(read_ahead > 0, "Read ahead should at least be 1.");
 
         SentProcessor {
             pipeline,
             writer,
             batch_size,
+            max_len,
             read_ahead,
             buffer: Vec::new(),
         }
     }
 
     pub fn process(&mut self, sent: Sentence) -> Result<(), Error> {
+        if let Some(max_len) = self.max_len {
+            // Length includes the root node.
```

Comment:
You could add this as a note to the error message in ```server.rs```, e.g. "Cannot parse maximum sentence length (incl. root)". As a user, I would not have thought of the root being included in the sentence length.

Reply:
It should not ;). There is a bug in the next line (off-by-2) and the comment is not very clear. Thanks for catching this, I have amended the PR.